crates/server/src/main.rs

```diff
@@ -30,13 +30,14 @@ async fn main() -> Result<()> {
         )
         .route("/", get(handlers::root::root))
         .route(
-            "/pricing_data",
-            post(handlers::get_pricing_data::get_pricing_data)
+            "/job_status/:job_id",
+            get(handlers::job_status::get_job_status)
                 .layer(from_fn_with_state(db.clone(), simple_apikey_auth)),
         )
         .route(
-            "/callback_test",
-            post(handlers::pricing_callback::pricing_callback),
+            "/pricing_data",
+            post(handlers::get_pricing_data::get_pricing_data)
+                .layer(from_fn_with_state(db.clone(), simple_apikey_auth)),
```

Comment:
should be able to apply the middleware for both of these by placing the `.layer` outside of the `route`. See https://docs.rs/axum/latest/axum/routing/struct.Router.html#method.layer

Reply:
Didn't know this, thanks for the tip

Reply:
@cwkang1998 this should be solved with the latest changes